src/components/TodoModal/TodoModal.tsx

```diff
@@ -15,25 +32,35 @@
               className="modal-card-title has-text-weight-medium"
               data-cy="modal-header"
             >
-              Todo #2
+              {`Todo #${activeTodo.id}`}
             </div>
 
             {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-            <button type="button" className="delete" data-cy="modal-close" />
+            <button
+              type="button"
+              className="delete"
+              data-cy="modal-close"
+              onClick={() => setActiveTodo(null)}
```

Comment:
You're directly manipulating the DOM by setting `setActiveTodo` to null in your `onClick` event handler. In React, it's better to avoid direct DOM manipulation as much as possible. Consider using a more React-like approach, such as setting state or using context.